storm-core/src/jvm/org/apache/storm/daemon/supervisor/timer/RunProfilerActions.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.daemon.supervisor.timer;
+
+import org.apache.storm.Config;
+import org.apache.storm.cluster.IStormClusterState;
+import org.apache.storm.daemon.supervisor.SupervisorData;
+import org.apache.storm.daemon.supervisor.SupervisorUtils;
+import org.apache.storm.generated.ProfileAction;
+import org.apache.storm.generated.ProfileRequest;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class RunProfilerActions implements Runnable {
+    private static Logger LOG = LoggerFactory.getLogger(RunProfilerActions.class);
+
+    private Map conf;
+    private IStormClusterState stormClusterState;
+    private String hostName;
+
+    private String profileCmd;
+
+    private SupervisorData supervisorData;
+
+    private class ActionExitCallback implements Utils.ExitCodeCallable {
+        private String stormId;
+        private ProfileRequest profileRequest;
+        private String logPrefix;
+
+        public ActionExitCallback(String stormId, ProfileRequest profileRequest, String logPrefix) {
+            this.stormId = stormId;
+            this.profileRequest = profileRequest;
+            this.logPrefix = logPrefix;
+        }
+
+        @Override
+        public Object call() throws Exception {
+            return null;
+        }
+
+        @Override
+        public Object call(int exitCode) {
+            LOG.info("{} profile-action exited for {}", logPrefix, exitCode);
+            try {
+                stormClusterState.deleteTopologyProfileRequests(stormId, profileRequest);
```

Comment:
The original code only did this if it was a stop.

```
                      action-on-exit (fn [exit-code]
                                       (log-message log-prefix " profile-action exited for code: " exit-code)
                                       (if stop?
                                         (delete-topology-profiler-action storm-cluster-state storm-id (thriftify-profile-request pro-action))))
```

Now it looks like we do it every time.  @kishorvpatil what are the ramifications of this?  Does it need to be added back in?